test/tests/functional/pbs_cray_check_node_exclusivity.py

```diff
@@ -0,0 +1,560 @@
+                           id=resv_node)
+        # waith 5 sec for job to end
+        self.server.expect(NODE, {'state': 'resv-exclusive'},
+                           id=resv_node, offset=5, interval=10)
+        # Wait for reservation to end and verify node state
+        # changed as free
+        msg = "Que;" + rid_q + ";deleted at request of pbs_server@"
+        self.server.log_match(msg, starttime=now, interval=2)
+        self.server.expect(NODE, {'state': 'free'},
+                           id=resv_node)
+
+    def test_job_inside_exclusive_reservation_on_compute_node(self):
+        """
+        Test Job will run correctly inside the exclusive
+        reservation requesting compute_node
+        """
+        if 'PBS_TZID' in self.conf:
+            tzone = self.conf['PBS_TZID']
+        elif 'PBS_TZID' in os.environ:
+            tzone = os.environ['PBS_TZID']
+        else:
+            self.logger.info('Missing timezone, using America/Los_Angeles')
+            tzone = 'America/Los_Angeles'
+        # Submit a standing reservation with place=excl
+        now = int(time.time())
+        a = {ATTR_l + '.select': '1:ncpus=1:vntype=cray_compute',
+             'Resource_List.place': 'excl',
+             ATTR_resv_rrule: 'FREQ=HOURLY;COUNT=1',
+             ATTR_resv_timezone: tzone,
+             'reserve_start': now + 10,
+             'reserve_end': now + 300}
+        r = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(r)
+        rid_q = rid.split('.')[0]
+        a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, a, id=rid)
+        node = self.server.status(RESV, 'resv_nodes', id=rid)
+        resv_node = self.server.reservations[rid].get_vnodes()[0]
+        self.server.expect(NODE, {'state': 'free'}, id=resv_node)
+        self.logger.info('Waiting 10s for reservation to start')
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, offset=10)
+        self.server.expect(NODE, {'state': 'resv-exclusive'},
+                           id=resv_node)
+        # Submit a job inside the reservation
+        submit_dir = self.du.create_temp_dir(asuser=TEST_USER)
+        a = {ATTR_q: rid_q}
+        j1 = Job(TEST_USER, attrs=a)
+        j1.create_script(self.script)
+        jid1 = self.server.submit(j1, submit_dir=submit_dir)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.expect(NODE, {'state': 'job-exclusive,resv-exclusive'},
+                           id=resv_node)
+        # waith 5 sec for job to end
+        self.server.expect(NODE, {'state': 'resv-exclusive'},
+                           id=resv_node, offset=5, interval=10)
+
+    def test_reservation_request_exclhost(self):
+        """
+        Test Reservation asking for place=exclhost on HostA
+        will have all resources of the vnodes present
+        on HostA assigned to it
+        """
```

Comment:
This test isn't very interesting since we ship PBS for cray ALPS with only one vnode per compute node.  This test is just like the other ones requesting excl.